lib/sles4sap/ipaddr2.pm

```diff
@@ -1915,4 +1946,247 @@ sub ipaddr2_test_other_vm {
     }
 }
 
+=head2 ipaddr2_clean_network_peering
+
+    ipaddr2_clean_network_peering(ibsm_rg => );
+
+Cleanup the network peering if needed.
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_clean_network_peering {
+    my (%args) = @_;
+    my $ibs_mirror_resource_group = $args{'ibsm_rg'};
+    my $source_group = ipaddr2_azure_resource_group();
+
+    qesap_az_vnet_peering_delete(source_group => $source_group, target_group => $ibs_mirror_resource_group);
```

Comment:
More on this: the whole function can be a single line (after checking `$args{ibsm_rg}` contains something). I would suggest:
```
sub ipaddr2_clean_network_peering {
    my (%args) = @_;
    croak 'Resource group of the IBS Mirror is required' unless $args{ibsm_rg};
    qesap_az_vnet_peering_delete(source_group => ipaddr2_azure_resource_group(), target_group => $args{ibsm_rg});
}
```